### PR TITLE
Post thumbnail support

### PR DESCRIPTION
**Pull request by brocheafoin**
This fixes issue #5 :

- Thumbnail image copied into the aggregating blog's Media library, attached to synced post
- Copied image set as post thumbnail for the corresponding post
- If thumbnail is changed or removed in subsite, change is replicated on aggregating blog

Could be coded more cleanly. All the image stuff in its own class, for example, but I'm pretty sure my code is closer to the coding guidelines than what's already in there, so I'm OK with the current state of affairs. I would really like this to be merged, though, so I'll do what you want. Cheers!

Oh! I forgot. One or two methods were ripped from[ Automattic's aggregator](https://github.com/Automattic/aggregator), so contributors list should be modified accordingly.

### CODE REVIEW

**Comment by Darthmaul** [comment]
This is awesome! Great work! I will be testing this ASAP! :D

**Comment by tw2113** [comment]
For information sake, see my comment at https://github.com/WebDevStudios/WDS-Multisite-Aggregate/issues/5#issuecomment-262440588 for how, to certain degrees, featured images aren't completely broken with this plugin. They're just handled differently.

**Comment by brocheafoin** [comment]
Hi Michael,

> featured images aren't completely broken with this plugin

I did not say that and if that's how you understood my message, I am really sorry. I had read the thread you point to before starting work. What it indicated to me was that people expected post thumbnails to be copied over to the aggregating blog. I know that's what I expected too. So I made it.

I understand there is some form of support, but it comes with a lot of drawbacks:

- Need to write custom code in your theme to display thumbnails
- Custom code needs to differentiate between aggregated and non-aggregated posts
- By default, only one thumbnail size available
- Need to write custom code to have access to more thumbnail sizes
- Even then, thumbnail sizes are those of the aggregated blog, not the aggregating blog

What I'm proposing here removes all these drawbacks. You can now use WP native functions in the aggregating blog for all posts, no matter where they come from. Thumbnails are available in all sizes defined in the aggregating blog.

Now, I've probably been a bit agressive in removing the meta fields used in the current method. I'll add them back for backwards compatibility. Would that help?

**Comment by tw2113** [comment]
Understood, and I wasn't offended or anything, just wanting to make sure it's clear to anyone coming by that may read it :)

I would definitely say strive for backwards compatibility as much as possible at this point.

Regarding some of the points you bring up and try to address, i agree and don't agree. The amount of custom code isn't as great as I think you perceived. Since it was stored as meta data on the aggregated posts, showing things would be as simple as `echo get_post_meta( ... );` and with the new setup, `the_post_thumbnail();` I will agree that with many themes coming with the_post_thumbnail() already ready and in place, it's merely less needing done immediately..

**Comment by brocheafoin** [comment]
I agree that not a lot of custom code is needed, but why require custom code at all when it's not necessary? Also, there is still the issue of the image sizes themselves, which may differ between the aggregated and the aggregating blogs. Agree to disagree? ;-)

I'll add back the current thumbnail meta fields to the aggregated posts. If I do this, will you consider merging this?

File: wds-multisite-aggregate.php
Comment (tw2113) [inline wds-multisite-aggregate.php]: Matching indent please.

File: wds-multisite-aggregate.php
Comment (tw2113) [inline wds-multisite-aggregate.php]: Matching indent please.

File: wds-multisite-aggregate.php
Comment (tw2113) [inline wds-multisite-aggregate.php]: Matching indent please.

File: wds-multisite-aggregate.php
Comment (tw2113) [inline wds-multisite-aggregate.php]: Matching indent please for the trailing }

**Comment by tw2113** [comment]
Gave some quick nitpick comments for some minor details.

@mmcachran when you get a chance, can you look over this PR and see if anything stands out for you?

Other than that, I have no present problem with merging it in. Would love for some of the people commenting in the issue to give it a test run as well.

**Comment by brocheafoin** [comment]
- Fixed the indentation issues (IDE was configured for spaces and not tabs).
- Added back the meta fields currently used for post thumbnails

> One or two methods were ripped from Automattic's aggregator, so contributors list should be modified accordingly.

The methods I copied and modified were all written by @philipjohn, if I'm to believe the `blame`.

**Comment by tw2113** [comment]
Still wanting someone besides just me to do a quick code review this before we merge in, for what it's worth.

**Comment by 10phillyphan** [comment]
Hey, I don't have much coding experience so I wouldn't be able to contribute much at all, but I was wondering if you (@brocheafoin ) could add support for scheduled posts. Your code for the plugin file works perfectly when pushing publish on a multisite, but when it is schedule for, say, 10am, the thumbnail will not transfer to the main blog. It works perfectly when straight-out pushing "Publish", but just not when scheduling. 

Any chance you can help out with that?

**Comment by brocheafoin** [comment]
Hi @10phillyphan, I wouldn't mind taking a look at it, but I'm also waiting for this pull request to be merged into the main branch. @tw2113, you were looking for more feedback; @10phillyphan here says they've been using this branch successfully. Does that count as enough feedback? Thanks!

**Comment by 10phillyphan** [comment]
@brocheafoin awesome, I appreciate that! Yes, it definitely works for me. No other plugin does exactly what I need it to do, but this one is as close to perfect as it can be. The scheduled post not showing the thumbnail issue is the only thing I still need to be fixed.

**Comment by tw2113** [comment]
For what it's worth, you should be able to pull theirs into yours, without us merging anything first. They would just want to do the PR against yours. Perhaps we could get both merged in at once instead of multiple PRs.

just some thoughts.

**Review by bibiwan — APPROVED** [review]
it's working :)

**Comment by streiten** [comment]
was just done fixing #8 with pr #18 when i saw the progress here nice to see this advance!

**Comment by tw2113** [comment]
@streiten is your PR still needed then? or did it end up getting covered here?

**Comment by streiten** [comment]
for legacy support i guess. when testing the new plugin version with old posts it wasnt covered.

**Pull request merged**
